tables/Tables/PlainTable.cc

```diff
@@ -285,12 +285,19 @@ void PlainTable::closeObject()
 	//# Check if table can indeed be deleted.
 	//# If not, set delete flag to False.
         //# It only checks if the main table is multi-used.
-	if (isMultiUsed(False)) {
-	    unmarkForDelete (False, "");
-	    throw (TableError ("Table " + name_p + " cannot be deleted;"
-			       " the table or a subtable is still used"
-			       " in another process"));
-	}
+        //# File locking support in Lustre (maybe other file systems too)
+        //# seems to be asynchronous to some degree, so try a few times.
+        int nTrys = 5;
+        while (isMultiUsed(False)) {
+            if (nTrys == 0) {
+                unmarkForDelete (False, "");
+                throw (TableError ("Table " + name_p + " cannot be deleted;"
+                                   " the table or a subtable is still used"
+                                   " in another process"));
+            }
+            sleep(1);
```

Comment:
Thanks. Good point.
I got the fix from somebody else.


On Wed, Aug 26, 2015 at 3:41 PM, Julian Taylor <notifications@github.com>
wrote:

> In tables/Tables/PlainTable.cc
> <https://github.com/casacore/casacore/pull/223#discussion_r37982157>:
>
> > -	    unmarkForDelete (False, "");
> > -	    throw (TableError ("Table " + name_p + " cannot be deleted;"
> > -			       " the table or a subtable is still used"
> > -			       " in another process"));
> > -	}
> > +        //# File locking support in Lustre (maybe other file systems too)
> > +        //# seems to be asynchronous to some degree, so try a few times.
> > +        int nTrys = 5;
> > +        while (isMultiUsed(False)) {
> > +            if (nTrys == 0) {
> > +                unmarkForDelete (False, "");
> > +                throw (TableError ("Table " + name_p + " cannot be deleted;"
> > +                                   " the table or a subtable is still used"
> > +                                   " in another process"));
> > +            }
> > +            sleep(1);
>
> sleep has poor interaction with signals, please use nanosleep
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/casacore/casacore/pull/223/files#r37982157>.
>